service/src/stage_worker.rs

```diff
@@ -139,23 +177,13 @@ async fn run_stage_task(
                     }
                 }
                 if stage.is_error() {
-                    let get_status = || {
-                        if stage.split_task.state != TASK_STATE_SUCCESS {
-                            crate::stage_service::stage_service::Status::SplitError
-                        } else {
-                            for task in &stage.prove_tasks {
-                                if task.state != TASK_STATE_SUCCESS {
-                                    return crate::stage_service::stage_service::Status::ProveError;
-                                }
-                            }
-                            if stage.agg_all_task.state != TASK_STATE_SUCCESS {
-                                crate::stage_service::stage_service::Status::AggError
-                            } else if stage.final_task.state != TASK_STATE_SUCCESS {
-                                return crate::stage_service::stage_service::Status::FinalError;
-                            } else {
-                                return crate::stage_service::stage_service::Status::InternalError;
-                            }
-                        }
+                    let get_status = || match stage.step {
```

Comment:
wrap the Steps, and implement the `get_status` as its method.

Reply:
The error here is defined in the protocol stage and does not want to be coupled with the protocol here